components/site/src/lib.rs

```diff
@@ -799,19 +799,30 @@ impl Site {
     }
 
     fn index_for_lang(&self, lang: &str) -> Result<()> {
-        let index_json = search::build_index(lang, &self.library.read().unwrap(), &self.config)?;
-        let (path, content) = match &self.config.search.index_format {
-            IndexFormat::ElasticlunrJson => {
-                let path = self.output_path.join(format!("search_index.{}.json", lang));
-                (path, index_json)
+        let extension = match &self.config.search.index_format {
```

Comment:
that should be a method on the `IndexFormat` enum rather than putting it here

Reply:
Moved the whole filename logic inside IndexFormat